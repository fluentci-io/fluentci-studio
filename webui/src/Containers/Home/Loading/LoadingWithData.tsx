import { FC, useEffect, useState } from "react";
import Loading from "./Loading";

const LoadingWithData: FC = () => {
  const [pkgxReady, setPkgxReady] = useState(false);
  const [denoReady, setDenoReady] = useState(false);
  const [hideDeno, setHideDeno] = useState(true);
  const [hidePkgx, setHidePkgx] = useState(true);

  useEffect(() => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (window as any).ipcRenderer.on(
      "setup-pkgx",
      (data: { message: string; done: boolean }) => {
        setPkgxReady(data.done);
        if (!data.done) {
          setHidePkgx(false);
        }
      }
    );
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (window as any).ipcRenderer.on(
      "setup-deno",
      (data: { message: string; done: boolean }) => {
        setDenoReady(data.done);
        if (!data.done) {
          setHideDeno(false);
        }
      }
    );
  }, []);

  return (
    <Loading
      pkgxReady={pkgxReady}
      denoReady={denoReady}
      hideDeno={hideDeno}
      hidePkgx={hidePkgx}
    />
  );
};

export default LoadingWithData;
