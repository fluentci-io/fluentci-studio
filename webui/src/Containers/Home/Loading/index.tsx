import Loading from "./LoadingWithData";

export default Loading;
