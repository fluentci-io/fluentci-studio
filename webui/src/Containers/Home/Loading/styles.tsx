import styled from "@emotion/styled";

export const Container = styled.div`
  margin: 0 auto;
  margin-top: calc(50vh - 100px);
  width: 400px;
`;

export const Line = styled.div`
  display: flex;
  flex-direction: row;
  height: 50px;
  align-items: center;
  justify-content: center;
`;
