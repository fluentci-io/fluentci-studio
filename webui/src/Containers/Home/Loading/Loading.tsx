import { FC } from "react";
import { Spinner } from "baseui/spinner";
import { CheckCircle } from "@styled-icons/boxicons-solid";
import { Container, Line } from "./styles";

export type LoadingProps = {
  pkgxReady: boolean;
  denoReady: boolean;
  hidePkgx?: boolean;
  hideDeno?: boolean;
};

const Loading: FC<LoadingProps> = (props) => {
  const { pkgxReady, denoReady, hideDeno, hidePkgx } = props;
  return (
    <Container>
      <div>
        {!hidePkgx && (
          <Line>
            {pkgxReady && (
              <CheckCircle
                size={22}
                style={{ marginRight: 15 }}
                color="#24ffd7"
              />
            )}
            {!pkgxReady && (
              <Spinner
                $size={"15px"}
                $borderWidth={"3px"}
                style={{
                  borderRightColor: "#ffffff22",
                  borderLeftColor: "#ffffff22",
                  borderTopColor: "#ffffff22",
                  borderBottomColor: "#24ffd7",
                  marginRight: 15,
                }}
              />
            )}
            <div>Setting up pkgx ...</div>
          </Line>
        )}

        {!hideDeno && (
          <Line>
            {denoReady && (
              <CheckCircle
                size={22}
                style={{ marginRight: 15 }}
                color="#24ffd7"
              />
            )}
            {!denoReady && (
              <Spinner
                $size={"15px"}
                $borderWidth={"3px"}
                style={{
                  borderRightColor: "#ffffff22",
                  borderLeftColor: "#ffffff22",
                  borderTopColor: "#ffffff22",
                  borderBottomColor: "#24ffd7",
                  marginRight: 15,
                }}
              />
            )}
            <div>Setting up Deno ...</div>
          </Line>
        )}
      </div>
    </Container>
  );
};

Loading.defaultProps = {
  hidePkgx: true,
  hideDeno: true,
};

export default Loading;
