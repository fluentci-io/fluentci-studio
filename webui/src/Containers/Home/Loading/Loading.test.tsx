import { render } from "@testing-library/react";
import Loading from "./Loading";

describe("Loading", () => {
  it("should render the component", () => {
    const { container } = render(<Loading pkgxReady={true} denoReady={true} />);
    expect(container).toMatchSnapshot();
  });
});
