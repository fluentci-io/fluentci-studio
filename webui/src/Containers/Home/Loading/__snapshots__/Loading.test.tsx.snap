// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading > should render the component 1`] = `
<div>
  <div
    class="css-1dwv9t7"
  >
    <div />
  </div>
</div>
`;
